mta/widgetastic/__init__.py

```diff
@@ -93,141 +80,68 @@ def delete_application(self, app_name):
-
-        @property
-        def description(self):
-            """return description of project"""
-            return self.browser.text(self.DESCRIPTION_LOCATOR)
-
-        @property
-        def application_count(self):
-            """return application count text"""
-            return self.browser.text(self.APPLICATION_COUNT)
-
-        @property
-        def last_updated(self):
-            """return last project updated info"""
-            return self.browser.text(self.LAST_UPDATED)
-
-        def delete(self):
-            """click on delete project"""
-            self.browser.click(self.DELETE_PROJECT)
-
-        def edit(self):
-            """click on edit project"""
-            self.browser.click(self.EDIT_PROJECT)
-
-        def select(self):
-            """click on specific project"""
-            self.browser.click(self.TITLE_LOCATOR)
-
-        def __repr__(self):
-            return f"Project({self.name})"
-
-    @property
-    def items(self):
-        """return all project names"""
-        return [
-            self.browser.text(self.TITLE_LOCATOR, parent=item)
-            for item in self.browser.elements(self.LIST_ITEM_LOCATOR)
-        ]
-
-    def read(self):
-        return self.items
-
-    @property
-    def projects(self):
-        """return all available project objects"""
-        return [self.project(item) for item in self.items]
+    """This is custom widget represent project list and provide actions to project"""
+
+    ACTIONS_INDEX = 4
+    table = PatternflyTable(
+        ".//table[contains(@class, 'pf-c-table')]",
+        column_widgets={
+            "Name": Text(locator=".//a"),
+            "Applications": Text(locator=".//td[@data-label='Applications']"),
+            "Status": Text(locator=".//td[@data-label='Status']"),
+            "Description": Text(locator=".//td[@data-label='Description']"),
+            ACTIONS_INDEX: Dropdown(),
+        },
+    )
 
     def get_project(self, name):
-        """ Get specific project
-        Args:
-            name: name of project
-        Returns:
-            list of project objects
-        """
-        if name in self.items:
-            return self.project(name)
-        else:
-            raise ProjectNotFound(f"Project {name} not found.")
+        for row in self.table:
+            if row.name.text == name:
+                return row
+
+    def exists(self, name):
+        for row in self.table:
+            if row.name.text == name:
+                return True
+        return False
+
+    def delete(self, name):
+        """click on delete project"""
+        for row in self.table:
+            if row.name.text == name:
+                row[self.ACTIONS_INDEX].widget.item_select("Delete")
+
+    def edit(self, name):
+        """click on edit project"""
+        for row in self.table:
+            if row.name.text == name:
+                row[self.ACTIONS_INDEX].widget.item_select("Edit")
```

Comment:
Optional:
I think rather than taking `name` i.e. project nave every time.. better to create project obj and add this functionality as part of it.

We can use the same class `class project(ParametrizedView)` and replace all operations on that specific table row.